src/he/he_internal.h

```diff
@@ -170,6 +171,7 @@ struct he_conn {
   /// SNI Hostname
   char sni_hostname[HE_MAX_HOSTNAME_LENGTH + 1];
 
+  /// Authentication data for both HE_AUTH_TYPE_TOKEN and HE_AUTH_TYPE_CB
```

Comment:
`auth_buffer` is overloaded to serve two types of auth, but it can only serve one type of auth as a time right?  If yes, then I would prefer we describe this as:

`/// Authentication data for either HE_AUTH_TYPE_TOKEN or HE_AUTH_TYPE_CB`